### ISSUE TITLE
 [GTNH 2.5.1] ArrayIndexOutOfBoundsException in specific chunks

### ISSUE THREAD

**Issue by RawHav0kk** [GTNewHorizons/Angelica#39]
### Your GTNH Discord Username

rawhav0kk

### Mod Version

1.0.0-alpha1

### Java Version

Java 19

### Bug Report

https://mclo.gs/CwxxcNF
Certain chunks, presumably that heavly use forge microblocks cause heavy rendering issues. Here is example rendering issue:
![rendering_issue](https://github.com/GTNewHorizons/Angelica/assets/75432277/2c0540a5-25cf-40f4-b558-2fd64c4225b0)

### Mod List

AdvancedSolarPanel-1.7.10-3.5.1a.jar
adventurebackpack-1.0.17-GTNH.jar
ae2fc-1.1.74-gtnh.jar
ae2stuff-0.6.0-GTNH.jar
AFSU-1.7.10-1.2.6-GTNH.jar
alchgrate-1.0.3-GTNH.jar
amazingtrophies-1.1.4.jar
AmunRa-GC-0.5.0.jar
angelica-1.0.0-alpha1.jar
AngerMod-1.7.10-0.6.3.jar
AppleCore-3.2.12.jar
appliedenergistics2-rv3-beta-292-GTNH.jar
ArchitectureCraft-1.8.6.jar
Aroma1997Core-1.7.10-1.0.2.16.jar
AromaBackup-1.7.10-0.1.0.0.jar
AsieLib-0.5.4.jar
Automagy-1.7.10-0.28.2.jar
Avaritia-1.46.jar
Avaritiaddons-1.6.0-GTNH.jar
backpack-2.2.12-GTNH.jar
bartworks-0.8.23.jar
battlegear2-1.3.0.jar
Baubles-1.0.3.jar
bdlib-1.7.10-1.9.8-GTNH.jar
beebetteratbees-0.3.1-GTNH.jar
BetterAchievements-1.7.10-0.1.3.jar
BetterBuildersWands-0.10.1-GTNH.jar
bettercrashes-1.3.5-GTNH.jar
betterloadingscreen-1.5.3-GTNH.jar
betterp2p-1.1.20.jar
BetterQuesting-3.4.7-GTNH.jar
BiblioCraft[v1.11.7][MC1.7.10].jar
BiblioWoods[BiomesOPlenty][v1.9].jar
BiblioWoods[Forestry][v1.7].jar
BiblioWoods[Natura][v1.5].jar
binnie-mods-2.2.4.jar
BiomesOPlenty-1.7.10-2.1.0.2308-universal.jar
BlockLimiter-1.7.10-0.55.jar
blockrenderer6343-1.0.6.jar
BloodArsenal-1.2.11.jar
BloodMagic-1.4.3.jar
Botania-1.10.3-GTNH.jar
BotanicHorizons-1.0.19-GTNH.jar
BrandonsCore-1.7.10-1.0.0.13-GTNH.jar
bugtorch-1.2.12-GTNH.jar
buildcraft-7.1.38.jar
buildcraft-compat-7.1.16.jar
BuildCraftOilTweak-1.7.10-1.0.4.jar
CarpentersBlocks-3.4.1-GTNH.jar
catwalks-1.7.10-2.1.4-GTNH.jar
chisel-2.12.3-GTNH.jar
chiseltones-1.7.10-1.0.4-GTNH.jar
CodeChickenCore-1.2.0.jar
CoFHCore-[1.7.10]3.1.4-329.jar
CompactKineticGenerators-1.7.10-1.0.jar
Computronics-1.7.1-GTNH.jar
controlling-2.0.1.jar
CookingForBlockheads-1.2.16-GTNH.jar
CraftPresence-2.2.5-1.7.10.jar
CraftTweaker-3.2.13.jar
creativecore-1.3.31-GTNH.jar
CropLoadCore-1.7.10-0.1.10.jar
CropsPP-1.5.12.jar
custommainmenu-1.10.3.jar
defaultkeys-1.1.6.jar
defaultserverlist-1.7.10-1.4.0.jar
defaultworldgenerator-1.7.10-0.2.jar
Draconic-Evolution-1.2.1-GTNH.jar
DummyCore-1.17.0.jar
duradisplay-1.1.7.jar
EMT-1.3.8.jar
endercore-0.2.19.jar
EnderIO-2.5.9.jar
EnderStorage-1.7.10-1.4.12.jar
EnderZoo-1.7.10-1.0.23.jar
EnhancedLootBags-1.1.0.jar
eternalsingularity-1.1.2.jar
extrautilities-1.2.13a.jar
findit-1.1.0.jar
FloodLights-1.7.10-1.2.9.jar
Forbidden.Magic-0.6.7-GTNH.jar
Forestry-4.7.1.jar
forgelin-1.9.7-GTNH.jar
ForgeMultipart-1.4.1.jar
ForgeRelocation-1.7.10-0.0.3.jar
ForgeRelocationFMP-1.7.10-0.0.4.jar
FTBLib-1.0.19-GTNH.jar
FTBUtilities-1.1.1-GTNH.jar
gadomancy-1.2.0.jar
galacticgreg-1.0.10.jar
Galacticraft-3.0.75-GTNH.jar
GalaxySpace-1.2.15-GTNH.jar
gendustry-1.7.10-1.6.5.5-GTNH.jar
ggfab-0.3.9.jar
GoodGenerator-0.7.17.jar
GraviSuite-1.7.10-2.0.3.jar
gravisuiteneo-1.0.20.jar
gregtech-5.09.44.110.jar
gtneioreplugin-1.1.3.jar
GTNewHorizonsCoreMod-2.2.55.jar
gtnhintergalactic-1.2.9.jar
gtnhlanth-0.11.9.jar
gtnhlib-0.0.13.jar
GTNH-TC-Wands-1.3.1.jar
GT-PlusPlus-1.10.54.jar
GT.Scanner.Mod-1.7.2.jar
GTTweaker-2.0.5.jar
HardcoreDarkness-MC1.7.10-1.7.jar
HardcoreEnderExpansion-1.7.10-1.9.7-GTNH.jar
harvestcraft-1.1.4-GTNH.jar
Healer-1.2.1.jar
HelpFixer-1.7.10-1.1.0.jar
hodgepodge-2.4.3.jar
holoinventory-2.3.2-GTNH.jar
HungerOverhaul-1.7.10-1.0.4-GTNH.jar
hydroenergy-1.1.1.jar
IC2 Crop Plugin Version 1.7.10-1.3.1.jar
IC2NuclearControl-2.5.1.jar
ifu-1.9.6.jar
IguanaTweaksTConstruct-2.3.0.jar
industrialcraft-2-2.2.828a-experimental.jar
InfernalMobs-1.7.9-GTNH.jar
InGameInfoXML-2.8.5.jar
INpureCore-1.1.5-GTNH.jar
inventorytweaks-1.6.1.jar
IronChest-6.0.74.jar
IronChestMinecarts-1.7.10-1.0.8.jar
irontanks-1.7.10-1.2.6.jar
JABBA-1.3.1.jar
journeymap-1.7.10-5.1.4p6-fairplay.jar
JustEnoughCalculation-1.7.10-4.0.jar
kekztech-0.9.6.jar
kubatech-0.13.12.jar
littletiles-1.2.9-GTNH.jar
logisticspipes-1.0.8-GTNH.jar
lootgames-1.7.10-2.0.8.jar
LunatriusCore-1.7.10-1.1.7-GTNH.jar
lwjgl3ify-1.5.7.jar
magicbees-2.7.1-GTNH.jar
malisiscore-0.14.9.jar
malisisdoors-1.14.0-GTNH.jar
Mantle-0.3.7.jar
mobsinfo-0.1.13-GTNH.jar
ModTweaker2-0.9.10.jar
modularui-1.1.24.jar
MouseTweaks-2.4.9-GTNH.jar
mxrandom-0.2.0.jar
Natura-2.5.7.jar
naturescompass-1.7.10-1.3.6-GTNH.jar
NEIAddons-1.13.0.jar
NEICustomDiagram-1.5.14.jar
NEIIntegration-1.3.3.jar
netherportalfix-1.7.10-1.1.2.jar
NodalMechanics-1.7.10-1.1.-6-GTNH.jar
NotEnoughEnergistics-1.4.6.jar
notenoughIDs-1.5.3.jar
NotEnoughItems-2.4.13-GTNH.jar
nutrition-0.0.5.jar
oauth-1.7.10-1.06.1-GTNH.jar
OpenBlocks-1.8.3-GTNH.jar
OpenComputers-1.9.21-GTNH.jar
OpenGlasses-1.4.2-GTNH.jar
OpenModsLibs-1.7.10-0.10.6.jar
OpenModularTurrets-1.7.10-2.2.11-247.jar
openprinter-1.7.10-0.1.3-GTNH.jar
opensecurity-1.7.10-1.0.120-GTNH.jar
Opis-1.3.9-mapless.jar
overloadedarmorbar-1.0.3.jar
Pam's Harvest the Nether 1.7.10a.jar
personalspace-1.0.28.jar
PlayerAPI-1.4.3.jar
ProjectBlue-1.1.12-GTNH.jar
ProjRed-4.8.1-GTNH.jar
Railcraft-9.15.3.jar
RandomThings-2.4.6.jar
RemoteIO-1.7.10-2.4.8.jar
roguelike-1.7.10-1.5.3-GTNH.jar
RWG-alpha-1.3.9-pre.jar
Schematica-1.9.4-GTNH.jar
SGCraft-1.7.10-1.3.13-GTNH.jar
sharewhereiam-2.0.2.jar
sleepingbag-1.7.10-0.1.4.jar
SpecialMobs-3.4.3.jar
SpiceOfLife-2.1.1-carrot.jar
SSP-1.1.2-GT-NH-Mod.jar
StevesAddons-1.7.10-0.10.27.jar
StevesCarts-1.7.10-2.0.2.jar
StevesFactoryManager-1.1.7-GTNH.jar
StorageDrawers-1.12.2-GTNH.jar
StorageDrawersBop-1.7.10-1.11.17-GTNH.jar
StorageDrawersForestry-1.7.10-1.11.17-GTNH.jar
StorageDrawersMisc-1.7.10-1.11.18-GTNH.jar
StorageDrawersNatura-1.7.10-1.11.17-GTNH.jar
structurecompat-0.4.0.jar
structurelib-1.2.9.jar
SuperTic-1.7.10-1.2.5.jar
Tainted-Magic-7.6.3-GTNH.jar
tcinventoryscan-1.7.10-1.0.12-GTNH.jar
tcneiadditions-1.2.2.jar
tcnodetracker-1.7.10-1.1.7.jar
TConstruct-1.10.13-GTNH.jar
TecTech-5.3.23.jar
TGregworks-1.7.10-GTNH-1.0.25.jar
Thaumcraft-1.7.10-4.2.3.5a.jar
Thaumcraft4Tweaks-1.5.18.jar
ThaumcraftMobAspects-1.7.10-1.0.0-GTNH.jar
thaumcraftneiplugin-1.7.10-1.7a.jar
ThaumcraftResearchTweaks-1.0.6.jar
Thaumic-Based-1.5.6.jar
thaumicboots-1.1.0.jar
thaumicenergistics-1.5.4-GTNH.jar
Thaumic-Exploration-1.2.0-GTNH.jar
ThaumicHorizons-1.4.2.jar
thaumicinsurgence-0.2.7.jar
Thaumic Machina-1.7.10-0.2.1.jar
ThaumicTinkerer-2.8.5.jar
TiCTooltips-1.3.0.jar
Tinkers-Defense-1.2.1d.jar
TMechworks-0.3.0.jar
TooMuchLoot-1.7.10-4.1.0-GTNH.jar
torohealth-1.0.4.jar
Translocator-1.1.2.21.jar
TravellersGear-1.7.10-1.16.6.jar
travellersgearneo-1.0.jar
TwilightForest-2.5.1.jar
txloader-1.6.3.jar
+unimixins-all-1.7.10-0.1.14.jar
Universal-Singularities-8.6.7.jar
visualprospecting-1.2.1.jar
Waila-1.6.5.jar
WailaHarvestability-1.7.10-1.1.10-GTNH.jar
WAILAPlugins-0.3.0.jar
WanionLib-1.8.4.jar
WarpTheory-1.2.17-GTNH.jar
Wawla-1.7.10-1.1.3-GTNH.jar
WirelessCraftingTerminal-1.10.1.jar
witchery-1.7.10-0.24.1.jar
WitcheryExtras-1.1.14.jar
WitchingGadgets-1.3.6-GTNH.jar
worldedit-v0.0.5.jar
WR-CBE-1.7.10-1.4.8.jar
YAMCore-0.5.86.jar
Ztones-1.7.10-2.2.2.jar

### Final Checklist

- [X] I have searched the issues and haven't found a similar issue.
- [X] I have read the known incompatibilities and this is not related to one of those.

**Comment by mitchej123** [comment]
Can you narrow down which block(s) are causing this, and try reproducing it?

**Comment by RawHav0kk** [comment]
> Can you narrow down which block(s) are causing this, and try reproducing it?

Specifically, it is a cyan-stained clay slab, in a vertical position. I will attempt to recreate it.

**Comment by RawHav0kk** [comment]
I got it! here is a schematic that if placed, causes the out of bounds exepction on any world:
[debug.zip](https://github.com/GTNewHorizons/Angelica/files/13767664/debug.zip)
Though, I am not sure what causes it now, maybe lanterns? Check it out for yourself

**Comment by RawHav0kk** [comment]
I narrowed it to extra utilities colored lamps, they are themselves transparent and make other things around them transparent.

**Comment by RawHav0kk** [comment]
@mitchej123 Hey, could you confirm if the information provided is detailed enough for the issue? Should I open an issue with a proper title, and close this one?
https://mclo.gs/QWtEWVy here is an extra log from client.

**Comment by mitchej123** [comment]
We've already identified the issue, and we've got a potential fix.

The renderer for exu is accessing a TE at y=256, and the same X/Y in each chunk.  We're rendering subchunks (16x16x16) + 2 extra blocks on each side, so it's getting an OOB when trying to find y=256.  It's a ISBRH which we render on the main thread, unless whitelisted, so we can pass it the full client world instead of a world slice....
